### PR TITLE
added restore deleted apps and get list of deleted apps

### PR DESCRIPTION
**Pull request by connorlee77**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by marstr — CHANGES_REQUESTED** [review]

**Comment by connorlee77** [comment]
Could @amarzavery @lmazuel @yugangw-msft test it out/fix any errors?

**Comment by marstr** [comment]
I'm going to be OOF all next week, so this PR should be reassigned.

File: specification/graphrbac/data-plane/stable/1.6/graphrbac.json [hunk 1]
```diff
@@ -148,6 +148,104 @@
...
+        "operationId": "DeletedApplications_Restore",
+        "description": "Restores the deleted application in the directory.",
+        "parameters": [
+    	  {
+            "name": "objectId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Application object ID."
+          },
```
Comment (lmazuel) [inline specification/graphrbac/data-plane/stable/1.6/graphrbac.json]: Probably missing the api version to get apiversion=1.6
```
          {
            "$ref": "#/parameters/ApiVersionParameter"
          },
```

File: specification/graphrbac/data-plane/stable/1.6/graphrbac.json [hunk 2]
```diff
@@ -148,6 +148,104 @@
...
+    },
+    "/{tenantID}/deletedApplications": {
+      "get": {
+        "tags": [
+          "deletedApplications"
+        ],
+        "operationId": "DeletedApplications_Get",
+        "description": "Gets a list of deleted applications in the directory.",
+        "parameters": [
+          {
```
Comment (lmazuel) [inline specification/graphrbac/data-plane/stable/1.6/graphrbac.json]: Missing the api version to get apiversion=1.6
```
          {
            "$ref": "#/parameters/ApiVersionParameter"
          },
```

**Review by lmazuel — CHANGES_REQUESTED** [review]

File: specification/graphrbac/data-plane/stable/1.6/graphrbac.json [hunk 3]
```diff
@@ -148,6 +148,104 @@
...
+          }
+        }
+      }
+    },
+    "/{tenantID}/deletedApplications/{applicationObjectId}": {
+      "delete": {
+        "tags": [
+          "Application"
+        ],
+        "operationId": "Applications_HardDelete",
```
Comment (lmazuel) [inline specification/graphrbac/data-plane/stable/1.6/graphrbac.json]: If I understand correctly the purpose of this operation, the idea is to "purge" deletedApplications and remove for real the application without waiting 30 days. This means the app has to have been deleted before.
If that so, then operationId should be `DeletedApplications_Delete`

**Comment by lmazuel** [comment]
Left the company.

**Pull request closed without merging**